cmd/sandbox-ce/compose/compose.go

```diff
@@ -204,3 +218,104 @@ func expandHome(path string) (string, error) {
 
 	return strings.Replace(path, "~", hd, 1), nil
 }
+
+// InitWorkdir creates a working directory in ~/.srcd for the given repositores
+// directory. The working directory will contain a docker-compose.yml and a
+// .env file.
+// If the directory is already initialized the function returns with no error
+func InitWorkdir(reposdir string) (string, error) {
+	err := initDefaultConfig()
+	if err != nil {
+		return "", err
+	}
+
+	homedir, err := os.UserHomeDir()
+	if err != nil {
+		return "", errors.Wrap(err, "could not detect home directory")
+	}
+
+	workdir := filepath.Join(homedir, ".srcd", "workdirs", reposdir)
+
+	err = os.MkdirAll(workdir, 0755)
+	if err != nil {
+		return "", errors.Wrap(err, "could not create working directory")
+	}
+
+	composePath := filepath.Join(workdir, "docker-compose.yml")
+	_, err = os.Stat(composePath)
```

Comment:
need a check for `err != nil && !os.IsNotExist(err)`

Reply:
good catch, done in 5667c969